Documentation/coding-guidelines/project-guidelines.md

```diff
@@ -1,49 +1,105 @@
+#Build Project Guidelines
+In order to work in corefx repo you must first run build.cmd/sh from the root of the repo at least
+once before you can iterate and work on a given library project.
+
+##Behind the scenes with build.cmd
+
+- Setup tools (currently done in init-tools but will later be a boot-strap script in run.cmd/sh)
+- Restore external dependencies
+ - CoreCLR - Copy to bin\runtime\$(TargetGroup)
+ - Netstandard Library - Copy to bin\ref\netstandard
+ - UAP - Copy to bin\runtime\uap??
+ - NetFx targeting pack - Copy to bin\ref\netfx
+- Build targeting pack
+ - Build src\ref.builds file which will build all the references assemblies to bin\ref\$(TargetGroup). Where TargetGroup is a versionless TFM represented by the BuildConfiguration (e.g. netcoreapp/netfx/uap)
+- Build product
+ - Build src\src.builds file which will build all the source projects
+ - Source projects will have `<Reference>` to their dependencies or in some
+   cases P2P references to other source projects when necessary (i.e. partial facades)
+ - Source projects will copy output to bin\runtime\$(TargetGroup)
+- Sign product
+ - Build src\sign.builds - TODO: We should make this as part of the src.builds file instead of a separate .builds file.
+
+##Behind the scenes with build-test.cmd
+- build-test.cmd cannot be ran successfully until build.cmd has been ran at least once for a BuildConfiguration.
+- Tests will be built against the bin\ref\$(TargetGroup) or bin\ref\netstandard, depending on the supported configurations in the test project.
+- Tests will be run against bin\runtime\$(TargetGroup) which is the live bits built from
+- TODO: Currently as part of src/post.builds we call CloudBuild.targets which setup our test runs. This needs to be moved to be part of build-test.cmd now.
+
+##Behind the scenes with build-packages.cmd
+- build-packages.cmd cannot be run successfully until build.cmd has been ran at least once for a BuildConfiguration.
+- Build src\packages.builds which will build only the packages it knows has the context to build which will generally be only the ones for the given BuildConfiguration. If a package requires assets from multiple BuildConfigurations it will require that all BuildConfigurations are built first.
+
+#Build Pivots
+Below is a list of all the various options we pivot the project builds on:
+
+- **Architecture:** x86, x64, ARM, ARM64, AnyCPU
+- **OS:** Windows_NT, Linux, OSX, FreeBSD, AnyOS
+- **Target Frameworks:** NetFx (aka Desktop), netstandard (aka dotnet/Portable), NETCoreApp (aka .NET Core), UAP (aka UWP/Store/netcore50)
```

Comment:
In my mind that falls under netcoreapp-aot. Do you think that will work or do you believe there will be something different?